src/main/java/frc/robot/commands/dropping/AutoAlignCommand.java

```diff
@@ -0,0 +1,237 @@
+            case B1 -> elevatorStateToPlacePosition(currentElevatorState, PlacePosition.B2_HYBRID, PlacePosition.B2_MID, PlacePosition.B2_HIGH);
+            case B2 -> elevatorStateToPlacePosition(currentElevatorState, PlacePosition.B3_HYBRID, PlacePosition.B3_MID, PlacePosition.B3_HIGH);
+            case B3 -> elevatorStateToPlacePosition(currentElevatorState, PlacePosition.C1_HYBRID, PlacePosition.C1_MID, PlacePosition.C1_HIGH);
+            case C1 -> elevatorStateToPlacePosition(currentElevatorState, PlacePosition.C2_HYBRID, PlacePosition.C2_MID, PlacePosition.C2_HIGH);
+            case C2 -> elevatorStateToPlacePosition(currentElevatorState, PlacePosition.C3_HYBRID, PlacePosition.C3_MID, PlacePosition.C3_HIGH);
+            default -> throw new RuntimeException("Position not found!");
+        });
+    }
+
+    /**
+     * Returns a provided HYBRID, MID, or HIGH place position, depending on the given elevator state.
+     * @param elevatorState The current elevator state of the robot.
+     * @param hybridPosition The HYBRID position to return.
+     * @param midPosition The MID position to return.
+     * @param highPosition The HIGH position to return.
+     * @return The provided HYBRID, MID, or HIGH place position, depending on the given elevator state.
+     */
+    private static PlacePosition elevatorStateToPlacePosition(
+        ElevatorState elevatorState, PlacePosition hybridPosition, PlacePosition midPosition, PlacePosition highPosition
+    ) {
+        return switch (elevatorState) {
+            case CONE_MID, CUBE_MID -> midPosition;
+            case CONE_HIGH, CUBE_HIGH -> highPosition;
+            default -> hybridPosition;
+        };
+    }
+
+    /**
+     * Gets the closest place position given the current robot position and extension of the elevator. Positions are
+     * initially compared with their distance to the robot, and ties (eg. between HIGH and MID positions) are broken
+     * using elevator extension.
+     * 
+     * @param robotPose The current pose of the robot.
+     * @param elevatorExtensionMeters The current extension, in meters, of the elevator.
+     * @param isRed Whether the robot is on the red team.
+     * @return The closest `PlacePosition`.
+     */
+    public static PlacePosition getClosestPlacePosition(Pose2d robotPose, double elevatorExtensionMeters, boolean isRed) {
+        return Collections.min(
+            Arrays.asList(PlacePosition.values()),
+            Comparator.comparing((PlacePosition pos) -> {
+                Translation2d targetTranslation = pos.alignPosition.getPose(isRed).getTranslation();
+                return robotPose.getTranslation().getDistance(targetTranslation);
+            }).thenComparing((PlacePosition pos) -> {
+                double targetExtensionMeters =  pos.elevatorState.getExtension(OffsetState.DROPPING, true);
+                return Math.abs(targetExtensionMeters - elevatorExtensionMeters);
```

Comment:
can we keep the current target elevator state instead? if the elevator is travelling to another state, it doesn't make sense to me to just take the closest one instead of getting to existing target